File: dev/front/src/types/knowledge.ts
/**
 * Knowledge 资源类型镜像(唯一事实源:API-CONTRACT.md §2.3)。
 */

/** GET /knowledge 单条文档。 */
export interface KDoc {
  /** 32 位 hex(uuid4().hex)。 */
  doc_id: string
  /** 上传时的原始文件名。 */
  filename: string
  /** ISO 8601,带时区偏移。 */
  created_at: string
  /** 该文档切块数。 */
  chunks: number
}

/** GET /knowledge 响应(按 created_at 升序)。 */
export interface KDocListResponse {
  docs: KDoc[]
}

/**
 * POST /knowledge/upload 响应。
 * `created: false` 为内容判重命中(解析后文本 SHA-256 相同),复用已有 doc_id,
 * 不是错误——UI 提示"该文档已存在,已复用"。
 */
export interface KUploadResult {
  doc_id: string
  created: boolean
  name: string
}

/** DELETE /knowledge/{doc_id} 响应(B7 落地前,删除不存在的文档会 500 而非 404)。 */
export interface KDeleteResult {
  ok: boolean
  doc_id: string
}

File: src/api/routers/knowledge.py
"""知识库路由(模块 11 T3):GET /knowledge / POST /knowledge/upload / DELETE /knowledge/{doc_id}。

复用 04 的 RAGStore(与 REPL /kb 同一套实现);上传大小上限 20MB(DEV 坑表)。
"""
import os
import tempfile
from pathlib import Path

from fastapi import APIRouter, File, HTTPException, UploadFile

from tools.rag.store import RAGStore

router = APIRouter(prefix="/knowledge", tags=["knowledge"])

MAX_UPLOAD_BYTES = 20 * 1024 * 1024  # 20MB


@router.get("")
def list_knowledge():
    return {"docs": RAGStore().list_docs()}


@router.post("/upload")
def upload_knowledge(file: UploadFile = File(...)):
    content = file.file.read()
    if len(content) > MAX_UPLOAD_BYTES:
        raise HTTPException(status_code=400, detail="文件超过 20MB 上限")
    name = file.filename or "upload"
    suffix = Path(name).suffix
    with tempfile.NamedTemporaryFile(suffix=suffix, delete=False) as tmp:
        tmp.write(content)
        tmp_path = tmp.name
    try:
        doc_id, created = RAGStore().upload(Path(tmp_path), name)
    finally:
        os.unlink(tmp_path)
    return {"doc_id": doc_id, "created": created, "name": name}


@router.delete("/{doc_id}")
def delete_knowledge(doc_id: str):
    RAGStore().delete(doc_id)
    return {"ok": True, "doc_id": doc_id}
